hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterClientProtocol.java

```diff
@@ -1935,6 +1935,21 @@ public DatanodeInfo[] getSlowDatanodeReport() throws IOException {
     return rpcServer.getSlowDatanodeReport(true, 0);
   }
 
+  @Override
+  public String getEnclosingRoot(String src) throws IOException {
+    Path mountPath = new Path("/");
+    if (subclusterResolver instanceof MountTableResolver) {
+      MountTableResolver mountTable = (MountTableResolver) subclusterResolver;
+      if (mountTable.getMountPoint(src) != null) {
+        // unclear if this is the correct thing to do, probably depends on default mount point / link fallback
+        mountPath = new Path(mountTable.getMountPoint(src).getSourcePath());
```

Comment:
getMonutPoint can return null.

Reply:
yeah we check that in the if condition above